packages/zero-unplugin/src/index.ts

```diff
@@ -231,15 +231,16 @@ export const plugin = createUnplugin<PluginOptions, true>((options) => {
             }),
           )}`;
           return {
-            code: `import ${JSON.stringify(data)};\n${result.code}`,
+            // CSS import should be the last so that nested components produce correct CSS order injection.
+            code: `${result.code}\nimport ${JSON.stringify(data)};`,
             map: result.sourceMap,
           };
         }
         const cssId = `./${cssFilename}`;
         cssFileLookup.set(cssId, cssFilename);
         cssLookup.set(cssFilename, cssText);
         return {
-          code: `import ${JSON.stringify(`./${cssFilename}`)};\n${result.code}`,
+          code: `${result.code}\nimport ${JSON.stringify(`./${cssFilename}`)};`,
```

Comment:
Do we need to change this for vite as well?

Reply:
This line does not affect Vite at the moment because `zero-vite-plugin` is not using `zero-unplugin` but I can confirm that by importing CSS first, the order is wrong even with Vite.

```diff
- result.code += `\nimport ${JSON.stringify(cssFilename)};\n`;
+ result.code = `import ${JSON.stringify(cssFilename)};\n${result.code}`;
```
https://github.com/mui/material-ui/blob/2d43ad1e8bd0916f2b7ba5e5143990bec49cf7f0/packages/zero-vite-plugin/src/zero-vite-plugin.ts#L254

Reply:
Is there ways to create unit tests for this?

Reply:
@siriwatknp vite plugin and unplugin both kind of duplicate the same code. So ideally, any changes done in unplugin should also be done in vite plugin.

Reply:
@oliviertassinari We could have a snapshot test for it that verifies the built css file changes. I'll have to look into how we can have a proper unit test for the build tool itself.